File: Solutions/whiteWeed/Game Shop/2/Item.java
/**
 * 
 */
package whiteWeed.gameShop;

/**
 * <pre>
 * whiteWeed.gameShop
 *   |_Item
 * 
 * 1. 개요 : 
 * 2. 작성일 : 2017. 5. 18.
 * </pre>
 * 
 * @author				:  Weed
 * @version				:  1.0
 */
public class Item {
	protected String name, description;
	protected int weight, value;
	
	public Item(String name, String description, int weight, int value){
		this.name = name;
		this.description = description;
		this.weight = weight;
		this.value = value;
	}
	
	public void describe(){
		System.out.println("Name          = " + this.name);
		System.out.println("Description = " + this.description);
		System.out.println("Weight        = " + this.weight + " lbs");
		System.out.println("Value           = " + this.value + " gold coins");
		System.out.println();
	}
}

File: Solutions/Kamikoto66/GameShop4/Item.cpp
#include <iostream>
#include "Item.h"

Item::Item()
{
	_name = "";
	_description = "";
	_weight = 0;
	_value = 0;
}

Item::~Item()
{
}

Item::Item(string name, string description, int weight, int value)
	:_name(name), _description(description), _weight(weight), _value(value)
{
}

void Item::Describe()
{
	cout << "Name:" << _name << endl;
	cout << "Description: " << _description << endl;
	cout << "Weight: " << _weight << "lbs" <<endl;
	cout << "Value: " << _value << "gold coins" <<endl;
}
